### PR TITLE
EZP-30331: Cannot edit ezmedia properties after publishing

### PR DESCRIPTION
**Pull request by ViniTou**
| Question           | Answer
| ------------------ | ------------------
| **JIRA issue**     | [EZP-30331](https://jira.ez.no/browse/EZP-30331)
| **Bug/Improvement**| yes
| **New feature**    | no
| **Target version** | `7.3` 
| **BC breaks**      | no
| **Tests pass**     | yes
| **Doc needed**     | no

Note for QA:
Please check save content before publishing, as there was bug with in in the past and that changes part of solution.

**TODO**:
- [x] Implement feature / fix a bug.
- [x] Implement tests.
- [x] Fix new code according to Coding Standards (`$ composer fix-cs`).
- [x] Ask for Code Review.

### CODE REVIEW

File: eZ/Publish/Core/FieldType/BinaryBase/BinaryBaseStorage/Gateway/DoctrineStorage.php [hunk 1]
```diff
@@ -107,6 +107,27 @@ protected function setInsertColumns(QueryBuilder $queryBuilder, VersionInfo $ver
...
+     */
+    protected function setUpdateColumns(QueryBuilder $queryBuilder, VersionInfo $versionInfo, Field $field)
+    {
+        $queryBuilder
+            ->set('contentobject_attribute_id', ':fieldId')
+            ->set('filename', ':filename')
+            ->set('mime_type', ':mimeType')
+            ->set('original_filename', ':originalFilename')
+            ->set('version', ':versionNo')
+            ->setParameter(':fieldId', $field->id, PDO::PARAM_INT)
```
Comment (alongosz) [inline eZ/Publish/Core/FieldType/BinaryBase/BinaryBaseStorage/Gateway/DoctrineStorage.php]: Please use for a new code
```suggestion
            ->setParameter(':fieldId', $field->id, ParameterType::INTEGER)
```
to make Doctrine\DBAL 3.0 migration easier in the future.
(not sure if ParameterType is imported in this class, so you may need to do it manually).

File: eZ/Publish/Core/FieldType/BinaryBase/BinaryBaseStorage/Gateway/DoctrineStorage.php [hunk 2]
```diff
@@ -115,6 +136,55 @@ protected function setInsertColumns(QueryBuilder $queryBuilder, VersionInfo $ver
...
+                        $this->connection->quoteIdentifier('contentobject_attribute_id'),
+                        ':fieldId'
+                    ),
+                    $updateQuery->expr()->eq(
+                        $this->connection->quoteIdentifier('version'),
+                        ':versionNo'
+                    )
+                )
+            )
+            ->setParameter(':fieldId', $field->id, PDO::PARAM_INT)
```
Comment (alongosz) [inline eZ/Publish/Core/FieldType/BinaryBase/BinaryBaseStorage/Gateway/DoctrineStorage.php]:
```suggestion
            ->setParameter(':fieldId', $field->id, ParameterType::INTEGER)
```

File: eZ/Publish/Core/FieldType/BinaryBase/BinaryBaseStorage/Gateway/DoctrineStorage.php [hunk 3]
```diff
@@ -115,6 +136,55 @@ protected function setInsertColumns(QueryBuilder $queryBuilder, VersionInfo $ver
...
+            ->setParameter(':versionNo', $versionInfo->versionNo, PDO::PARAM_INT)
```
Comment (alongosz) [inline eZ/Publish/Core/FieldType/BinaryBase/BinaryBaseStorage/Gateway/DoctrineStorage.php]:
```suggestion
            ->setParameter(':versionNo', $versionInfo->versionNo, ParameterType::INTEGER)
```

File: eZ/Publish/Core/FieldType/Media/MediaStorage/Gateway/DoctrineStorage.php [hunk 4]
```diff
@@ -111,4 +111,35 @@ protected function setInsertColumns(QueryBuilder $queryBuilder, VersionInfo $ver
...
+            ->set('pluginspage', ':pluginsPage')
+            ->set('quality', ':quality')
+            ->set('width', ':width')
+            ->setParameter(':controls', '')
+            ->setParameter(
+                ':hasController',
+                $field->value->externalData['hasController'],
+                PDO::PARAM_INT
+            )
+            ->setParameter(':height', $field->value->externalData['height'], PDO::PARAM_INT)
```
Comment (alongosz) [inline eZ/Publish/Core/FieldType/Media/MediaStorage/Gateway/DoctrineStorage.php]:
```suggestion
            ->setParameter(':height', $field->value->externalData['height'], ParameterType::INTEGER)
```

File: eZ/Publish/Core/FieldType/Media/MediaStorage/Gateway/DoctrineStorage.php [hunk 5]
```diff
@@ -111,4 +111,35 @@ protected function setInsertColumns(QueryBuilder $queryBuilder, VersionInfo $ver
...
+            ->setParameter(':isAutoplay', $field->value->externalData['autoplay'], PDO::PARAM_INT)
```
Comment (alongosz) [inline eZ/Publish/Core/FieldType/Media/MediaStorage/Gateway/DoctrineStorage.php]:
```suggestion
            ->setParameter(':isAutoplay', $field->value->externalData['autoplay'], ParameterType::INTEGER)
```

File: eZ/Publish/Core/FieldType/Media/MediaStorage/Gateway/DoctrineStorage.php [hunk 6]
```diff
@@ -111,4 +111,35 @@ protected function setInsertColumns(QueryBuilder $queryBuilder, VersionInfo $ver
...
+            ->setParameter(':isLoop', $field->value->externalData['loop'], PDO::PARAM_INT)
```
Comment (alongosz) [inline eZ/Publish/Core/FieldType/Media/MediaStorage/Gateway/DoctrineStorage.php]:
```suggestion
            ->setParameter(':isLoop', $field->value->externalData['loop'], ParameterType::INTEGER)
```

File: eZ/Publish/Core/FieldType/Media/MediaStorage/Gateway/DoctrineStorage.php [hunk 7]
```diff
@@ -111,4 +111,35 @@ protected function setInsertColumns(QueryBuilder $queryBuilder, VersionInfo $ver
...
+            ->setParameter(':pluginsPage', '')
+            ->setParameter(':quality', 'high')
+            ->setParameter(':width', $field->value->externalData['width'], PDO::PARAM_INT)
```
Comment (alongosz) [inline eZ/Publish/Core/FieldType/Media/MediaStorage/Gateway/DoctrineStorage.php]:
```suggestion
            ->setParameter(':width', $field->value->externalData['width'], ParameterType::INTEGER)
```

**Comment by lserwatka** [comment]
You can merge it up.

**Pull request merged**